### ISSUE TITLE
Performance problems with the diff plugin

### ISSUE THREAD

**Issue by DaVarga** [paleite/eslint-plugin-diff#2]
Hello I came across this plugin today.
Looks very promising but I have some problems once i integrate this plugin into my eslintrc. The linting process takes about 10-20 times longer with the plugin. I have configured everything according to the readme.
Even if I disable almost all rules it takes forever to complete. Without the plugin a linting process for the whole project takes about 20 seconds. Even if I only want to link one file with the plugin it takes over a minute.

I am using Windows 10 with nodejs v12.16.1. Let me know if I can provide more information.

**Comment by amc1804** [comment]
I haven't tried running it, just skimmed the code. When eslint reports many messages for one file, is the plugin re-running `git diff` for each message? It should only need to run `git diff` and extract the line ranges once per file. (Or it could run `git diff` just once for all the files with messages, and parse the pathnames along with the line numbers.)

**Comment by DaVarga** [comment]
You are right, `getDiffForFile` is called once for each lint message. getDiffForFile itself spawns a child process.
https://github.com/paleite/eslint-plugin-diff/blob/2b66df4d8dc8e28cde801ba075fc9150369964fd/src/processors.ts#L17

Second thing to optimize would be to lint only files that have any changes. Currently all files are linted and the corresponding messages are removed.

**Comment by paleite** [comment]
Wow!

This has been on my todo-list since I first released this plugin, but I hadn't realized that it would make such a big difference in other OSes. I also didn't think of preprocessing the list and filtering out any files without diffs, so that was another great addition 🏆

I really appreciate you taking the time to report this and am even more grateful for your PR. I've looked through it and it looks very good. Great to see the updates to the tests as well!

I will merge and release a new version during the next day, so I hope you (and everyone else using this plugin) can benefit of the performance optimizations very soon.

🇩🇪 Bin dir sehr dankbar dass du dir so viel Mühe gegeben hast!
